fixSpec.py

```diff
@@ -0,0 +1,159 @@
+    '/projects',
+    '/projects/{id}',
+    '/projects/{id}/devices',
+    '/devices/{id}',
+    '/organizations',
+    '/organizations/{id}/projects',
+]
+
+
+def setup_yaml():
+    """ https://stackoverflow.com/a/8661021 """
+
+    def represent_dict_order(self, data): return self.represent_mapping(
+        'tag:yaml.org,2002:map', data.items())
+    yaml.add_representer(OrderedDict, represent_dict_order)
+
+
+setup_yaml()
+
+
+def fixNumInt(di):
+    if isinstance(di, dict):
+        for k, v in di.items():
+            if k == "format" and v == "integer":
+                di["type"] = "integer"
+                di["format"] = "int32"
+            fixNumInt(v)
+    elif isinstance(di, list):
+        for v in di:
+            fixNumInt(v)
+
+
+def fixExplode(di):
+    if isinstance(di, dict):
+        if 'explode' in di:
+            if 'items' in di['schema']:
+                pass
+            else:
+                del di['explode']
+                #ic("NOT OK", di)
+            return
+        for k, v in di.items():
+            fixExplode(v)
+    elif isinstance(di, list):
+        for v in di:
+            fixExplode(v)
+
+
+def findSchemas(di):
+    ret = set()
+
+    def walk(di):
+        if isinstance(di, dict):
+            for k, v in di.items():
+                walk(v)
+        elif isinstance(di, list):
+            for v in di:
+                walk(v)
+        else:
+            if (type(di) == str) and ("/schemas/" in di):
+                ret.add(di.split("/")[-1])
+    walk(di)
+    return ret
+
+
+def loadYaml(fn):
+    with open(fn) as f:
+        return yaml.load(f, Loader=FullLoader)
+
+
+originalSpec = loadYaml(INFILE)
+fixedSpec = copy.deepcopy(originalSpec)
+
+tags = []
+paths = OrderedDict()
+schemas = OrderedDict()
+
+for t in originalSpec['tags']:
+    if t['name'] in WANT_TAGS:
+        tags.append(t)
+
+for path in originalSpec['paths']:
+    if path in WANT_PATHS:
+        paths[path] = originalSpec['paths'][path]
+
+neededSchemas = findSchemas(paths)
+for s in neededSchemas:
+    schemas[s] = originalSpec['components']['schemas'][s]
+
+neededSchemas = set()
+newSchemasDiscovered = True
+
+# this is just cherrypicking schemas that are wanted
+while newSchemasDiscovered:
+    newSchemasDiscovered = False
+    newSchemas = findSchemas(schemas)
+    addSchemas = newSchemas - neededSchemas
+    neededSchemas = neededSchemas.union(newSchemas)
+    newSchemasDiscovered = len(addSchemas) > 0
+
+    for s in addSchemas:
+        if s in neededSchemas:
+            schemas[s] = originalSpec['components']['schemas'][s]
+
+fixedSpec['info']['description'] = "desc"
+del fixedSpec['x-tagGroups']
+fixedSpec['components']['schemas'] = schemas
+fixedSpec['components']['requestBodies'] = {}
+fixedSpec['tags'] = tags
+fixedSpec['paths'] = paths
+
+# FIX 0. organization href to project
+fixedSpec['components']['schemas']['Project']['properties']['organization'] = {
+    "$ref": "#/components/schemas/Organization"
```

Comment:
Would it make sense to use `oneOf` `ref:Href` or `ref:Organization`? 

The discriminator would be that an organization_id is required in an Organization. (I mean discriminator generically, I don't believe OAS3 discriminator would work here because there is no single field that says "Href" or "Organization" as a value).

Reply:
Thanks for pointing this out, it should be 

```
fixedSpec['components']['schemas']['Project']['properties']['organization'] = {
    "$ref": "#/components/schemas/Href"
```

.. just in the other linked resources

Reply:
I just fixed it in https://github.com/t0mk/equinixmetalpy/pull/1/commits/a15e49774c41bf8d8b237bc8f96cddd646c99d0b